packages/testing/src/TestUtils.ts

```diff
@@ -0,0 +1,149 @@
+ *
+ * @param seed {string} Input to generate the keypair from.
+ * @returns Object with secret and public key and the key type.
+ */
+export function makeEncryptionKeypair(seed: string): {
+  secretKey: Uint8Array
+  publicKey: Uint8Array
+  type: EncryptionKeyType
+} {
+  const { secretKey, publicKey } = naclBoxPairFromSecret(blake2AsU8a(seed, 256))
+  return {
+    secretKey,
+    publicKey,
+    type: EncryptionKeyType.X25519,
+  }
+}
+
+/**
+ * Generates a callback that can be used for encryption.
+ *
+ * @param secretKey The options parameter.
+ * @param secretKey.secretKey The key to use for encryption.
+ * @param secretKey.type The X25519 type, only this one is supported.
+ * @returns The callback.
+ */
+export function makeEncryptCallback({
+  secretKey,
+}: {
+  secretKey: Uint8Array
+  type: EncryptionKeyType.X25519
+}): NaclBoxCapable['encrypt'] {
+  return async function encryptCallback({ data, peerPublicKey, alg }) {
+    const { box, nonce } = Crypto.encryptAsymmetric(
+      data,
+      peerPublicKey,
+      secretKey
+    )
+    return { alg, nonce, data: box }
+  }
+}
+
+/**
+ * Generates a callback that can be used for decryption.
+ *
+ * @param secretKey The options parameter.
+ * @param secretKey.secretKey The key to use for decryption.
+ * @param secretKey.type The X25519 type, only this one is supported.
+ * @returns The callback.
+ */
+export function makeDecryptCallback({
+  secretKey,
+}: {
+  secretKey: Uint8Array
+  type: EncryptionKeyType.X25519
+}): NaclBoxCapable['decrypt'] {
+  return async function decryptCallback({ data, nonce, peerPublicKey, alg }) {
+    const decrypted = Crypto.decryptAsymmetric(
+      { box: data, nonce },
+      peerPublicKey,
+      secretKey
+    )
+    if (!decrypted) throw new Error('Decryption failed')
+    return { data: decrypted, alg }
+  }
+}
+
+/**
+ * Generates a callback that can be used for signing.
+ *
+ * @param keypair The keypair to use for signing.
+ * @returns The callback.
+ */
+export function makeSignCallback(keypair: KeyringPair): SignCallback {
+  return async function sign<A extends SigningAlgorithms>({
+    alg,
+    data,
+  }: SigningData<A>): Promise<ResponseData<A>> {
+    const signature = keypair.sign(data, { withType: false })
+    return { alg, data: signature }
+  }
+}
+
+const keypairTypeForAlg: Record<SigningAlgorithms, KeypairType> = {
+  [SigningAlgorithms.Ed25519]: 'ed25519',
+  [SigningAlgorithms.Sr25519]: 'sr25519',
+  [SigningAlgorithms.EcdsaSecp256k1]: 'ecdsa',
+}
+
+export interface KeyTool {
+  keypair: KeyringPair
+  sign: SignCallback
+  authenticationKey: NewLightDidAuthenticationKey
+}
+
+/**
+ * Generates a keypair usable for signing and a few related values.
+ *
+ * @param alg The algorithm to use for the keypair.
+ * @returns The keypair, matching sign callback, a key usable as DID authentication key.
+ */
+export function makeSigningKeypair(
+  alg: SigningAlgorithms = SigningAlgorithms.Sr25519
+): KeyTool {
```

Comment:
`makeEncryptionKeypair` returns just the keypair, whereas this function returns KeyTool, which includes more than that. Can we either make the return types consistent (e.g. by returning an encryption keytool from `makeEncryptionKeypair`, which includes the encrypt and decrypt callbacks) or change the naming of this function?

Reply:
I expected this question :) Should they become `makeSigningKeyTool` and `makeEncryptionKeyTool`?